functional_tests/steps/bundles.py

```diff
@@ -121,3 +127,131 @@ def the_user_can_see_the_bundle_details_with_creator(context: Context) -> None:
 def bundle_inspect_show(context: Context) -> None:
     expect(context.page.get_by_text("Created by")).to_be_visible()
     expect(context.page.get_by_text(context.bundle_creator.get_username())).not_to_be_visible()
+
+
+tomorrow = timezone.now() + timedelta(days=1)
+
+
+# To test release calendar page panel
+@given("a release calendar page with a future release date exists")
+@given('a release calendar page with a "{status}" status and future release date exists')
+def release_calendar_page_with_status_and_future_date_exists(context: Context, status: str = "Provisional") -> None:
+    context.release_calendar_page = ReleaseCalendarPageFactory(
+        release_date=tomorrow,
+        status=status.upper(),
+        notice="default notice",
+    )
+    context.release_calendar_page.save_revision().save()
+
+
+@step('the user manually creates a future release calendar page with a "{status}" status')
+def user_manually_creates_future_release_calendar(context: Context, status: str):
+    navigate_to_release_calendar_page(context)
+    click_add_child_page(context)
+    title = "Future Release Calendar Page"
+    context.page.get_by_placeholder("Page title*").fill(title)
+    context.page.get_by_label("Status*").select_option(status.upper())
+    formatted_date = tomorrow.strftime("%Y-%m-%d %H:%M")
+    context.page.get_by_role("textbox", name="Release date*").fill(formatted_date)
+    context.page.get_by_role("region", name="Summary*").get_by_role("textbox").fill("My example release page")
+
+    # Save the values to access later
+    context.release_calendar_page = {
+        "title": title,
+        "release_date_value": ons_date_format(tomorrow, "DATETIME_FORMAT"),
+        "status": status,
+    }
+
+
+@when("the user enters a title")
+def user_enters_title(context: Context):
+    context.page.get_by_role("textbox", name="Name*").fill("Test Bundles")
+
+
+@step("the user selects the existing release calendar page")
+def user_creates_bundle_with_future_release_calendar_page(context: Context) -> None:
+    try:
+        title = context.release_calendar_page.title
+    except AttributeError:
+        title = context.release_calendar_page["title"]
+
+    context.page.get_by_text(title).click()
```

Comment:
Sorry, just a thought for here... Might be better to save the values in a different property (line 159) and then check for the existence of the page property (`release_calendar_page`) or your new property (e.g. `saved_release_calendar_page_details`). This way the code will be less confusing as we can then safely assume that `release_calendar_page` is always a page object, if not `None`.

Otherwise looking at the code, you treat this both as a object and a dict, which looks confusing. Separate names might make things clear.